src/components/forms/FormSearchField.js

```diff
@@ -1,46 +1,96 @@
+    const filter = createFilterOptions();
+    const { t } = useTranslation("common");
+
+    React.useEffect(() => {
+        const val = {};
+        val[props.labelKey] = value;
+        setSearchValue(val);
+    }, [value, setSearchValue, props.labelKey]);
+
+    const onOptionSelected = (event, newValue) => {
+        if (newValue) {
+            if (typeof newValue === "string") {
+                setFieldValue(field.name, newValue);
+            } else if (newValue && newValue.inputValue) {
+                // Create a new value from the user input
+                setFieldValue(field.name, newValue.inputValue);
+            } else {
+                setFieldValue(field.name, newValue[props.valueKey]);
+            }
+        } else {
+            setFieldValue(field.name, "");
+        }
     };
 
     return (
-        <FormControl fullWidth error={!!errorMsg}>
-            <Autocomplete
-                label={label}
-                controlShouldRenderValue
-                isClearable={!required}
-                cacheOptions
-                defaultInputValue={value}
-                onChange={onOptionSelected}
-                onBlur={onSearchBlur}
-                {...field}
-                {...props}
-            />
-            <FormHelperText>{errorMsg || helperText}</FormHelperText>
-        </FormControl>
+        <Autocomplete
+            id="searchField"
+            getOptionSelected={(option, value) =>
+                option[props.labelKey] === value[props.labelKey]
+            }
+            getOptionLabel={(option) => (option ? option[props.labelKey] : "")}
+            options={options}
+            size={size}
+            freeSolo={true}
+            selectOnFocus={true}
+            value={searchValue}
+            blurOnSelect={true}
+            handleHomeEndKeys={true}
+            onInputChange={handleSearch}
+            onChange={onOptionSelected}
+            renderOption={(option, state) => renderCustomOption(option)}
+            filterOptions={(options, params) => {
+                const filtered = filter(options, params);
+                // Suggest the creation of a new value
+                if (params.inputValue !== "") {
+                    filtered.push({
+                        inputValue: params.inputValue,
+                        label: t("formatTermFreeTextOption", {
+                            inputValue: params.inputValue,
+                        }),
+                    });
+                }
+
+                return filtered;
+            }}
+            renderInput={(params) => (
+                <FormTextField
+                    label={label}
+                    variant="outlined"
+                    form={form}
+                    field={field}
+                    helperText={helperText}
+                    required={required}
+                    {...props}
+                    {...params}
+                />
```

Comment:
`FormTextField` already sets `variant="outlined"`, and now that all `props` are getting spread into this field, I think there's no longer a need to remove `label`, `helperText`, and `required` from the props in the destructuring above. If those props are omitted from the destructing, then they wouldn't have to be repeated here:
```suggestion
                <FormTextField
                    form={form}
                    field={field}
                    {...props}
                    {...params}
                />
```